### PR TITLE
443 message threads

### PR DESCRIPTION
**Pull request by jasonmule**
This pull request makes the following changes:
- Shows email/ SMS messages related to a post.
- Allow a user to reply to an incoming email/ SMS.

Test these changes by:
- pull email/ sms messages using `bin/ushahidi dataprovider incoming`
- Go to the Post detail of a new post created from a message and follow the link view messages associated with the post.
- You should be able to reply to messages and view them on a thread.

Fixes ushahidi/platform#443 .

Ping @ushahidi/platform

---
This change is [Reviewable](https://reviewable.io/reviews/ushahidi/platform-client/146)

### CODE REVIEW

File: app/post/controllers/post-messages-controller.js [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+            .then(function (message) {
+                $scope.post.sets = _.without($scope.post.sets, String(collectionId));
+                Notify.showNotificationSlider(message);
+            });
+        }, function (errorResponse) {
+            Notify.showApiErrors(errorResponse);
+        });
+    };
+    /*
+    scope.searchCollections = function (query) {
```
Comment (rjmackay) [inline app/post/controllers/post-messages-controller.js]: Drop the commented code?

File: server/www/templates/posts/detail.html [hunk 2]
```diff
@@ -61,7 +61,11 @@ <h1 class="page-title">
                        <p>
                           <strong class="icon-left fa-circle">
                               {{form_name || "Unstructured"}}
-                          </strong> posted at {{post.updated || post.created | date:'shortTime'}}<strong> {{post.updated || post.created | date:'mediumDate'}}</strong> <br> <span ng-if="post.user.realname != null"><translate>post.by</translate> <a href="/settings/users/{{post.user.id}}">{{post.user.realname}}</a></span>
+                          </strong> posted at {{post.updated || post.created | date:'shortTime'}}<strong> {{post.updated || post.created | date:'mediumDate'}}</strong> <br>
```
Comment (rjmackay) [inline server/www/templates/posts/detail.html]: I suspect this should all be a single translation string ie. "posted at {{time}} by {{source}} etc" .. but not sure how to handle the HTML in that.

File: server/www/templates/posts/detail.html [hunk 3]
```diff
@@ -61,7 +61,11 @@ <h1 class="page-title">
...
+                          <span ng-if="post.user.realname != null"><translate>post.by</translate> <a href="/settings/users/{{post.user.id}}">{{post.user.realname}}</a></span>
+                          <span ng-if="post.source == 'sms'" translate>post.messages.by_sms</span>
+                          <span ng-if="post.source == 'email'" translate>post.messages.by_email</span>
```
Comment (rjmackay) [inline server/www/templates/posts/detail.html]: Need to account for other sources: ie. twitter etc

**Comment by rjmackay** [comment]
Code looks pretty good - but I'll give it a proper test in the morning.
Is this include code from msheria? or a full rewrite?

**Comment by jasonmule** [comment]
This is mostly a merge from msheria.

File: app/post/controllers/post-messages-controller.js [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+            });
+    }
+
+    // Set the page title
+    $translate('post.message_thread').then(function (title) {
+        $scope.title = title;
+        $scope.$emit('setPageTitle', title);
+    });
+
+    ContactEndpoint.get({ id: $scope.post.contact.id })
```
Comment (rjmackay) [inline app/post/controllers/post-messages-controller.js]: The original code of `$scope.contact = ContactEndpoint.get({ id: $scope.post.contact.id });` should have worked here.
Angular uses some magic to put the result back on the $scope.contact when its loaded..
Either works, so just an FYI

**Comment by rjmackay** [comment]
Did some minor clean up and removed some copy paste code from msheria. This should be good to go now.

---

Reviewed 14 of 14 files at r1, 3 of 3 files at r2.
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/ushahidi/platform-client/146#-:-KEd99MB4LH-OUPJtJVB:1025820160)*

**Comment by rjmackay** [comment]
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[server/www/templates/posts/detail.html, line 67 \[r1\]](https://reviewable.io:443/reviews/ushahidi/platform-client/146#-KE2qv20xF2amWJ_OgJL-r1-67:-KEd9WbDT2eHBGH0SFC7:-1705654646) ([raw file](https://github.com/ushahidi/platform-client/blob/c27f371415197f364e779bc790a5a1e026683f7e/server/www/templates/posts/detail.html#L67)):*
I think this is the only outstanding issue ^

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/ushahidi/platform-client/146)*

File: app/post/controllers/post-messages-controller.js (hunk 4 above)
Comment (jasonmule) [inline app/post/controllers/post-messages-controller.js]: Noted. Thanks!

**Comment by rjmackay** [comment]
Reviewed 2 of 2 files at r3.
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/ushahidi/platform-client/146)*

**Comment by rjmackay** [comment]
Good to go :shipit: 

---

Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/ushahidi/platform-client/146#-:-KFuZn8hkiTlSHEQOt4h:1178353558)*

**Pull request merged**